packages/realm/src/app-services/App.ts

```diff
@@ -183,12 +193,18 @@ export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType
   constructor(configOrId: AppConfiguration | string) {
     const config: AppConfiguration = typeof configOrId === "string" ? { id: configOrId } : configOrId;
     assert.object(config, "config");
-    const { id, baseUrl, app, timeout, multiplexSessions = true } = config;
+    const { id, baseUrl, app, timeout, multiplexSessions = true, baseFilePath } = config;
     assert.string(id, "id");
     if (timeout !== undefined) {
       assert.number(timeout, "timeout");
     }
     assert.boolean(multiplexSessions, "multiplexSessions");
+    if (baseFilePath !== undefined) {
+      assert.string(baseFilePath, "baseFilePath");
+      fs.setDefaultDirectoryPath(baseFilePath);
```

Comment:
I don't think we should update the default path, just because someone choose to create an app instance with some random path?
```suggestion
```

Reply:
I see your point, and I think we need to find a "unified" way to set the default directory (one that works for both local and synced Realms) in the future. I'll change it so it `App.Config.baseFilePath` and `Realm.defaultDirectory` are two independent ways to specify where Realms are stored by default.